src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/InvokerUtil.cs

```diff
@@ -150,8 +166,81 @@ internal static InvokeDelegate GenerateInvokeDelegate(MethodInfo method, out int
+
+                List<ParameterExpression> variables = new();
+                var result = Expression.Variable(typeof(object), "result");
+                variables.Add(result);
+
+                List<ParameterExpression> outputVariables = new();
+                List<ParameterExpression> invocationParameters = new();
+                List<Expression> expressions = new();
+
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    Type variableType = parameters[i].ParameterType.IsByRef
+                        ? parameters[i].ParameterType.GetElementType()
+                        : parameters[i].ParameterType;
+                    ParameterExpression variable = Expression.Variable(variableType, $"p{i}");
+
+                    if (ServiceReflector.FlowsIn(parameters[i]))
+                    {
+                        expressions.Add(Expression.Assign(variable, Expression.Convert(Expression.ArrayIndex(inputsParam, Expression.Constant(inputParameterCount)), variableType)));
+                        inputParameterCount++;
+                    }
+
+                    if (ServiceReflector.FlowsOut(parameters[i]))
+                    {
+                        outputParameterCount++;
+                        outputVariables.Add(variable);
+                    }
+
+                    variables.Add(variable);
+                    invocationParameters.Add(variable);
+                }
+
+                var castTargetParam = Expression.Convert(targetParam, method.DeclaringType);
+
+                expressions.Add(returnsValue
+                    ? Expression.Block(Expression.Assign(result, Expression.Convert(Expression.Call(castTargetParam, method, invocationParameters), typeof(object))))
+                    : Expression.Block(Expression.Call(castTargetParam, method, invocationParameters),
+                            Expression.Assign(result, Expression.Constant(null, typeof(object)))));
+
+                int j = 0;
+                foreach (var outputVariable in outputVariables)
+                {
+                    expressions.Add(Expression.Assign(
+                        Expression.ArrayAccess(outputsParam, Expression.Constant(j)),
+                        Expression.Convert(outputVariable, typeof(object))));
+                    j++;
+                }
+
+                expressions.Add(Expression.Block(
+                    Expression.Empty(),
+                    result));
+
+                BlockExpression finalBlock = Expression.Block(
+                        variables: variables,
+                        expressions: expressions);
+
+                Expression<InvokeDelegate> lambda = Expression.Lambda<InvokeDelegate>(
+                    finalBlock,
+                    targetParam,
+                    inputsParam,
+                    outputsParam);
 
-                return lambda;
+                return lambda.Compile();
             }
```

Comment:
Can we add an ILogger with a special category to output the equivalent code to help diagnose any problems people might have. Log at the verbose level so most won't see it. See the [example code here](https://learn.microsoft.com/en-us/dotnet/api/system.linq.expressions.blockexpression?view=net-7.0#examples) to see what you should expect. I think you should be able to log what the equivalent code would be.

Reply:
done

Reply:
I need to figure out how to get an `ILogger` there as this code is not DI aware.

Reply:
@mconnew done! I'm bringing the ILogger through DispatchOperation

Reply:
Please see https://dev.azure.com/dotnet/CoreWCF/_build/results?buildId=98641&view=logs&j=d68afe5c-acae-5905-6bc4-d7e9ecfea404&t=2c035bf3-2a1f-5fc5-9550-cc6f70ae3aed&l=10893

Reply:
Can you add methodInfo.ToString() as well to the log output so we can see what the interface method is declared as?